src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -73,4 +76,46 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
                 return false;
         }
     }
+
+    public async checkAndConfigureFirewall(): Promise<void> {
+        const firewallRuleName: string = 'azureDatabasesForVSCode-publicIp';
+        const resourceGroup: string = azureUtils.getResourceGroupFromId(this.id);
+        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+        const serverName: string = nonNullProp(this.server, 'name');
+        const ip: string = await publicIp.v4();
+        let existingFirewallRule: FirewallRule | undefined;
+
+        // Creating/updating a firewall rule takes a long time. So check if the rule already exists and is valid before creating/updating
+        try {
+            existingFirewallRule = await client.firewallRules.get(resourceGroup, serverName, firewallRuleName);
+        } catch (error) {
+            if (parseError(error).errorType === 'ResourceNotFound') {
+                // This firewall rule is not configured yet. Ignore error
+            } else {
+                throw error;
+            }
+        }
+
+        if (!existingFirewallRule || existingFirewallRule.startIpAddress !== ip || existingFirewallRule.endIpAddress !== ip) {
+            await ext.ui.showWarningMessage(
+                localize('configureFirewall', 'Your IP ({0}) is not included in the firewall rules for "{1}". Would you like to add it?', ip, this.server.name),
+                { modal: true },
+                { title: DialogResponses.yes.title }
+            );
+
+            const newFirewallRule: FirewallRule = {
+                startIpAddress: ip,
+                endIpAddress: ip
+            };
+
+            const options: vscode.ProgressOptions = {
+                location: vscode.ProgressLocation.Notification,
+                title: localize('configuringFirewall', 'Adding firewall rule for IP "{0}" to server "{1}"...', ip, serverName)
```

Comment:
How long exactly does this take? Should we add text like "This may take several minutes" to give people an idea or is it shorter than that?

Reply:
This generally takes about 30 seconds, probably short enough to avoid that text